bukkit/src/main/java/hu/montlikadani/tablist/bukkit/tablist/groups/impl/ReflectionHandled.java

```diff
@@ -66,75 +71,82 @@ public void registerTeam(String teamName) {
 			infoList = (List<Object>) infoListField.get(packetPlayOutPlayerInfo);
 
 			for (TabListUser user : plugin.getUsers()) {
-				ReflectionUtils.sendPacket(user.getPlayer(), packetPlayOutPlayerInfo);
+				ReflectionUtils.sendPacket(user.getPlayer(), newTeamPacket);
 			}
+
 		} catch (Exception e) {
 			e.printStackTrace();
 		}
 	}
 
 	@Override
-	public void setTeam(String teamName) {
-		registerTeam(teamName);
-
-		try {
-			updateName(tabListUser.getGroupPlayer().getCustomTabName());
-		} catch (Exception e) {
-			e.printStackTrace();
-		}
+	public void setTeam(GroupPlayer groupPlayer) {
+		registerTeam(groupPlayer);
+		updateName(groupPlayer.getCustomTabName());
 	}
 
 	@Override
-	public void unregisterTeam(String teamName) {
-		if (packetPlayOutPlayerInfo == null) {
-			return;
-		}
-
+	public void unregisterTeam(GroupPlayer groupPlayer) {
 		try {
-			updateName(tabListUser.getPlayer().getName());
-		} catch (Exception e) {
-			e.printStackTrace();
+
+			String teamName = groupPlayer.getFullGroupTeamName();
+			scoreRef.init();
+
+			Object oldTeamPacket = scoreRef.getScoreboardTeamConstructor().newInstance();
```

Comment:
This will instantiates a new instances for every of the classes which causes to not remove the oldest one.

Reply:
Do you mean `scoreRef.init();`? Where will be "the oldest one"?

Reply:
Looking more closely at the code, I see that this `unregisterTeam` method is also called when we first register the groups, that’s correct, but what if we want to remove it but don’t know because `scoreRef.init()` will be called. To fix this, the `scoreRef.init` code must be passed to the `registerTeam` method before the `unregisterTeam` method can be called. Hope understandable.

Reply:
okey, now code is correct?